papermill/tests/test_iorw.py

```diff
@@ -106,12 +107,22 @@ def test_read(self):
 
     def test_write(self):
         """
-        Tests that the `write` function performs a request a put request to
-        the giving path.
+        Tests that the `write` function performs a put request to the given
+        path.
         """
         path = 'http://example.com'
         buf = '{"papermill": true}'
 
         with patch('papermill.iorw.requests.put') as mock_put:
             HttpHandler.write(buf, path)
             mock_put.assert_called_once_with(path, json=json.loads(buf))
+
+    def test_write_failure(self):
+        """
+        Tests that the `write` function raises on failure to put the buffer.
+        """
+        path = 'http://localhost:9999'
+        buf = '{"papermill": true}'
+
+        with self.assertRaises(ConnectionError) as e:
```

Comment:
could you use `pytest`'s `pytest.raises(ConnectionError)` context manager and match on the message including the failed path in question?

Reply:
I could, mostly I wanted to just prove that request exceptions are raised and not ignored -- I care less what their contents are

Reply:
If you want me to make that change I'll refine the test to be more specific

Reply:
Think it's ok to just make an issue for making it more specific about handling an exact message match?